apollo-router-core/src/spec/query.rs

```diff
@@ -666,6 +670,13 @@ impl Operation {
             kind,
         })
     }
+
+    fn is_introspection(&self) -> bool {
+        self.selection_set.iter().any(|sel| match sel {
+            Selection::Field { name, .. } => name.as_str().starts_with("__"),
```

Comment:
the name here can be an alias that does not contain `__`. Example: https://github.com/apollographql/router/blob/main/apollo-router-core/well_known_introspection_queries/bananacakepop-phase1.graphql#L2
`schema: __type(name: "__Schema") {`

Reply:
Regarding this documentation I don't think https://spec.graphql.org/October2021/#sec-Schema-Introspection

Reply:
About the name, it will always take the name and not the alias, so it's good, in your example we will have `__type...` and not `schema`

Reply:
Good catch, I thought the name was indeed the name but it's sometimes replaced by the alias. I fixed it here https://github.com/apollographql/router/pull/802/commits/d61f25d37c234e2a9eb0818dd963cb2afd395634#diff-9537771c4f15ed3c4048f89f52707deed837ceddd967f75421a4937017575670L66